pulsar/src/tests.rs

```diff
@@ -175,3 +177,48 @@ fn custom_log_dir_test() {
     #[cfg(target_os = "windows")]
     assert!(log_path.ends_with("AppData/Local/pulsar/logs"));
 }
+
+#[cfg(test)]
+mod create_or_move_data_tests {
+
+    use std::fs;
+
+    use super::*;
+    use crate::utils::data_dir_getter;
+
+    /// Ensuring the function correctly handles the case where the old and new
+    /// directories are the same.
+    #[test]
```

Comment:
I would prefer to test a scenario where we have actual files in old directory which we move into new and delete the old.